analyzers/tests/SonarAnalyzer.UnitTest/TestCases/ExistsInsteadOfAny.cs

```diff
@@ -0,0 +1,177 @@
+        intList.Any(x => 0 == x); // Compliant (should raise S6617)
+        intList.Any(x => x == someInt); // Compliant (should raise S6617)
+        intList.Any(x => someInt == x); // Compliant (should raise S6617)
+        intList.Any(x => x.Equals(0)); // Compliant (should raise S6617)
+        intList.Any(x => 0.Equals(x)); // Compliant (should raise S6617)
+
+        intList.Any(x => x == x); // Noncompliant
+        intList.Any(x => someInt == anotherInt); // Noncompliant
+        intList.Any(x => someInt == 0); // Noncompliant
+        intList.Any(x => 0 == 0); // Noncompliant
+
+        intList.Any(x => x.Equals(x)); // Noncompliant
+        intList.Any(x => someInt.Equals(anotherInt)); // Noncompliant
+        intList.Any(x => someInt.Equals(0)); // Noncompliant
+        intList.Any(x => 0.Equals(0)); // Noncompliant
+        intList.Any(x => x.Equals(x + 1)); // Noncompliant FP
+
+        intList.Any(x => x.GetType() == typeof(int)); // Noncompliant
+        intList.Any(x => x.GetType().Equals(typeof(int))); // Noncompliant FP
+        intList.Any(x => MyIntCheck(x)); // Noncompliant
+        intList.Any(x => x != 0);     // Noncompliant
+        intList.Any(x => x.Equals(0) && true);   // Noncompliant
+        intList.Any(x => (x == 0 ? 2 : 0) == 0); // Noncompliant
+        intList.Any(x => { return x == 0; }); // Noncompliant FP
+
+        stringList.Any(x => x == ""); // Compliant (should raise S6617)
+        stringList.Any(x => "" == x); // Compliant (should raise S6617)
+        stringList.Any(x => x == someString); // Compliant (should raise S6617)
+        stringList.Any(x => someString == x); // Compliant (should raise S6617)
+        stringList.Any(x => x.Equals("")); // Compliant (should raise S6617)
+        stringList.Any(x => "".Equals(x)); // Compliant (should raise S6617)
+        stringList.Any(x => Equals(x, "")); // Noncompliant FP
+
+        stringList.Any(x => MyStringCheck(x)); // Noncompliant
+        stringList.Any(x => x != "");     // Noncompliant
+        stringList.Any(x => x.Equals("") && true);   // Noncompliant
+        stringList.Any(x => (x == "" ? "a" : "b") == "a"); // Noncompliant
+        stringList.Any(x => x.Equals("" + someString)); // Noncompliant FP
+
+        intArray.Any(x => x == 0); // Noncompliant (this is not raising S6617)
+        intArray.Any(x => 0 == x); // Noncompliant (this is not raising S6617)
+        intArray.Any(x => x == someInt); // Noncompliant (this is not raising S6617)
+        intArray.Any(x => someInt == x); // Noncompliant (this is not raising S6617)
+        intArray.Any(x => x.Equals(0)); // Noncompliant (this is not raising S6617)
+        intArray.Any(x => 0.Equals(x)); // Noncompliant (this is not raising S6617)
+        intArray.Any(x => x.Equals(x + 1)); // Noncompliant (this is not raising S6617)
+
+        bool MyIntCheck(int x) => x == 0;
+        bool MyStringCheck(string x) => x == "";
+    }
```

Comment:
The testcases above this comment are excellent. 💥 
I would like to request one more thing: Add some of the cases that the rule should not raise because of `reference ==`. To be specific, I mean something like this:

```csharp
        intList.Any(x => x == 0); // Compliant (should raise S6617)
        intList.Any(x => 0 == x); // Compliant (should raise S6617)
        intList.Any(x => x == someInt); // Compliant (should raise S6617)
        intList.Any(x => someInt == x); // Compliant (should raise S6617)
```

...but instead of integer collection, make a collection of a reference type (they should all be noncompliant).

Almost there! :)

Reply:
Extra note: I think this is also why the coverage is failing, your lists are always valuetypes or strings, if I did not miss something.

Extra extra note: Maybe add one more testcase that accepts a Func, like `.Any(method)` (should be noncompliant)